src/mlpack/methods/ann/model_parser.hpp

```diff
@@ -0,0 +1,294 @@
+/**
+ * @file model_parser.hpp
```

Comment:
Should this be part of the PR?

Reply:
No this shouldn't have been. I'm a little new to branching, the old changes also got reflected in the new branch somehow. Got to figure that out.

Reply:
I see, what I usually do, is to branch from master. But that also means I don't do any work in the master branch.

Reply:
Looks like in your case you already did some changes in the master branch, so maybe it makes sense to first move those changes into another branch and remove the commits from the master branch.